src/Schemas.jl

```diff
@@ -112,44 +112,49 @@ function codom end
 # Basic Schemas
 ###############
 
-abstract type TypeLevelBasicSchema{Name, obs, homs, attrtypes, attrs} <: TypeLevelSchema{Name} end
+abstract type TypeLevelBasicSchema{Name, obs, homs, attrtypes, attrs, eqs} <: TypeLevelSchema{Name} end
 
-const TypeLevelBasicCSetSchema{Name, obs, homs} = TypeLevelBasicSchema{Name, obs, homs, Tuple{}, Tuple{}}
+const TypeLevelBasicCSetSchema{Name, obs, homs} = TypeLevelBasicSchema{Name, obs, homs, Tuple{}, Tuple{}, Tuple{}}
 
 @struct_hash_equal struct BasicSchema{Name} <: Schema{Name}
   obs::Vector{Name}
   homs::Vector{Tuple{Name,Name,Name}}
   attrtypes::Vector{Name}
   attrs::Vector{Tuple{Name,Name,Name}}
-  function BasicSchema{Name}(obs, homs, attrtypes, attrs) where {Name}
-    new{Name}(obs, homs, attrtypes, attrs)
+  eqs::Vector{Tuple{Union{Nothing, Name}, Name, Name, Tuple{Vararg{Tuple{Vararg{Name}}}}}}
+  function BasicSchema{Name}(obs, homs, attrtypes, attrs, eqs) where {Name}
+    new{Name}(obs, homs, attrtypes, attrs, eqs)
   end
-  function BasicSchema(obs::Vector{Name}, homs, attrtypes, attrs) where {Name}
-    new{Name}(obs, homs, attrtypes, attrs)
+  function BasicSchema(obs::Vector{Name}, homs, attrtypes, attrs, eqs=nothing) where {Name}
+    eqs = isnothing(eqs) ? Tuple{Name, Name, Name, Tuple{Vararg{Tuple{Vararg{Name}}}}}[] : eqs
+    new{Name}(obs, homs, attrtypes, attrs, eqs)
   end
-  function BasicSchema(obs::Vector{Name}, homs) where {Name}
-    new{Name}(obs, homs, Name[], Tuple{Name,Name,Name}[])
+  function BasicSchema(obs::Vector{Name}, homs, eqs=nothing) where {Name}
+    eqs = isnothing(eqs) ? Tuple{Name, Name, Name, Tuple{Vararg{Tuple{Vararg{Name}}}}}[] : eqs
+    new{Name}(obs, homs, Name[], Tuple{Name,Name,Name}[], eqs)
   end
   function BasicSchema{Name}() where {Name}
     new(
       Vector{Name}(),
       Vector{Tuple{Name,Name,Name}}(),
       Vector{Name}(),
-      Vector{Tuple{Name,Name,Name}}()
+      Vector{Tuple{Name,Name,Name}}(),
+      Vector{Tuple{Union{Nothing, Name}, Name, Name, Vector{Vector{Name}}}}()
```

Comment:
Perhaps we should document this format. From the tests, I'm guessing that these are:

- Name of equation/axiom (optional)
- Start of path
- End of path
- Vector of paths

I guess the last one is `Vector{Vector{Name}}` instead of `Pair{Vector{Name}}` because you want to allow more than two equations with the same source/target to be stored more succinctly?

Reply:
Yes, your intuitions are all correct, though I'm thinking for now the generality of n-ary equations is unnecessary at the moment, so I'll switch it to Pair!